nautobot/ui/src/views/generic/ObjectRetrieve.js

```diff
@@ -362,190 +44,45 @@ export default function ObjectRetrieve({ api_url }) {
-                                    <CardHeader>
-                                        <Heading
-                                            display="flex"
-                                            alignItems="center"
-                                            gap="5px"
-                                        >
-                                            <NtcThumbnailIcon
-                                                width="25px"
-                                                height="30px"
-                                            />{" "}
-                                            Change Log
-                                        </Heading>
-                                    </CardHeader>
-                                    <ObjectListTable
-                                        tableData={changelogData.results}
-                                        defaultHeaders={
-                                            changelogHeaderData.view_options
-                                                .list_display_fields
-                                        }
-                                        tableHeaders={
-                                            changelogHeaderData.view_options
-                                                .fields
-                                        }
-                                        totalCount={changelogData.count}
-                                        active_page_number={0}
-                                        page_size={50}
-                                        tableTitle={"Change Logs"}
-                                        include_button={false}
-                                        data_loaded={changelogDataLoaded}
-                                        data_fetched={!changelogDataFetching}
-                                    />
-                                </Card>
-                            </TabPanel>
-                        ) : (
-                            () => {}
-                        )}
-                    </TabPanels>
-                </Tabs>
-            </Box>
-        </GenericView>
-    );
-    let return_view = default_view;
     if (
         AppComponents.CustomViews?.[route_name] &&
         "retrieve" in AppComponents.CustomViews?.[route_name]
     ) {
         const CustomView = AppComponents.CustomViews[route_name].retrieve;
-        return_view = <CustomView {...obj} />;
+        return <CustomView {...data} />;
     }
 
-    return return_view;
+    // NOTE: This acts as a schema that would be gotten form an OPTIONS call,
+    // which loads the tabs and its layout schema.
+    const objectRetrieveTabSchema = {
+        tabs: {
+            Location: schemaData.view_options.retrieve,
+        },
+    };
+
+    return (
+        <GenericView objectData={data}>
+            <Box background="white-0" borderRadius="md">
+                <RenderHeader data={data} />
+                <RenderTabs
```

Comment:
Much better 👏 

Let's update the names to be more descriptive, something like `RetreiveViewHeader` and `RetreiveViewTabs`.